File: src/main/java/info/skyblond/archivedag/repo/ConfigRepository.java
package info.skyblond.archivedag.repo;

import info.skyblond.archivedag.model.entity.ConfigEntity;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface ConfigRepository extends JpaRepository<ConfigEntity, String> {

    Page<ConfigEntity> findAllByKeyStartingWith(String keyPrefix, Pageable pageable);

    ConfigEntity findByKey(String key);
}

File: backend/src/main/kotlin/info/skyblond/archivedag/arstue/repo/ConfigRepository.kt
package info.skyblond.archivedag.arstue.repo

import info.skyblond.archivedag.arstue.entity.ConfigEntity
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface ConfigRepository : JpaRepository<ConfigEntity, String> {

    fun findAllByKeyStartingWith(keyPrefix: String, pageable: Pageable): Page<ConfigEntity>

    fun findByKey(key: String): ConfigEntity?
}
